trade-finance-manager-ui/server/controllers/helpers/isFacilityEndDateEnabledForFacility.js

```diff
@@ -0,0 +1,13 @@
+import { isTfmFacilityEndDateFeatureFlagEnabled, MAPPED_FACILITY_TYPE } from '@ukef/dtfs2-common';
+
+/**
+ * Facilities are part of a GEF deal if and only if they have a cash or contingent facility type.
+ * In mapGefFacility.js, the received facility gets mapped from 'CASH' or 'CONTINGENT' to 'Cash facility' and 'Contingent facility'
+ */
+const isFacilityPartOfGefDeal = (type) => {
+  return type === MAPPED_FACILITY_TYPE.CASH || type === MAPPED_FACILITY_TYPE.CONTINGENT;
+};
+
+export const isFacilityEndDateEnabledForFacility = (facility) => {
+  return isTfmFacilityEndDateFeatureFlagEnabled() && isFacilityPartOfGefDeal(facility.facilitySnapshot.type);
```

Comment:
Just stumbled across this - I don't have a great deal of context - why do we need to specifically check that the facility is part of a GEF deal, as part of the "is the date feature flag enabled", could it just check that the feature flag is set?

Could we maybe instead add a `isGefFacility` flag to `mapGefFacility, to save checks further down the line such as here? Just thinking otherwise if we need to check like this in other TFM areas.


Otherwise, could we rename `isFacilityPartOfGefDeal` to `isGefFacility`? Keep it simple

Reply:
I think its a choice between repeating the following
```
const isFacilityEndDateEnabledForFacility = isTfmFacilityEndDateFeatureFlagEnabled() && facility.isPartOfGefDeal);
if (isFacilityEndDateEnabledForFacility) {
    ...
}
```

vs

```
if (isFacilityEndDateEnabledForFacility(facility)) {
    ...
}
```

I have no preference either way -- I'll work on the refactor and if you want it reverted, it can be reverted :)

Reply:
Thanks @AlexBramhill I think we should update `mapGefFacility` to have an `isGefFacility` flag - would like to look further though, I suspect there maybe other areas that are checking if it's GEF (Cash etc) facility, if we have it all in TFM's `mapGefFacility` then those checks only happen in one central place and makes for easy consumption, instead of adding an additional helper for specific use in conditions/functions like we have here currently?

I'm not understanding why this is called `isFacilityEndDateEnabledForFacility`, can it just be `isFacilityEndDateEnabled`?

I think this condition is the clearest IMO:

```ts
isTfmFacilityEndDateFeatureFlagEnabled() && facility.isGefFacility;
```

Another topic/PR, but maybe could change the structure for feature flag checks for consumption by other places, i'm thinking instead of `isTfmFacilityEndDateFeatureFlagEnabled` , maybe we could have something like `featureFlags.tfm.facility.endDate` - I think would be easier to follow etc if we have more feature flags moving forward. I think this needs a team chat though.

```ts
facility.isGefFacility && featureFlags.tfm.facility.endDate
```

Reply:
...Or maybe even uppercase in constants so it's clear it's a constant/environment type thing e.g `FEATURE_FLAGS.TFM.FACILITY.END_DATE`, but another topic/discussion 😄

Reply:
I've seperated this out as requested. As menntioned, it's not particularly dry due to us having to repeat this logic in places, however the FF does not belong on the object and it is not going to be around forever, so this is acceptable.